src/core/Model.js

```diff
@@ -5,9 +5,26 @@
 class Model {
 
   /**
```

Comment:
Just so I understand correctly, this still requires us to create Models, but it makes it easier to validate new properties against those Models. So as of right now, only MultiLangStrings and Languages could be passed through this function because no others exist yet.

Reply:
Not 100% sure I'm answering your question, but:

Yes, right now we can only use the `defineModelProp()` method to define properties whose values are required to be Languages or MultiLangStrings (or other native classes like Map, Set, etc.).

Properties like `Language.name` and `Text.title` etc. always need to be MultiLangStrings. Previously, every time we wanted to define a MultiLangString property on a model, we'd have to repeat similar code that validates the values of these properties and makes sure those values are transformed into MultiLangStrings. This utility function just creates an abstraction for that code, keeping our codebase DRY.

So now, anytime you need to define a MultiLangString property on a model (or any property whose value is always a certain type of model), you can just call `Model.defineModelProp()` instead of repeating all the necessary code.